astronomer/providers/google/cloud/triggers/gcs.py

```diff
@@ -87,3 +89,245 @@ async def _object_exists(self, hook: GCSHookAsync, bucket_name: str, object_name
             else:
                 res = "pending"
             return res
+
+
+class GCSPrefixBlobTrigger(GCSBlobTrigger):
+    """
+     A trigger that fires and it looks for all the objects in the given bucket
+     which matches the given prefix if not found sleep for certain interval and checks again.
+
+    :param bucket: the bucket in the google cloud storage where the objects are residing.
+    :param prefix: The prefix of the blob_names to match in the Google cloud storage bucket
+    :param google_cloud_conn_id: reference to the Google Connection
+    :param polling_period_seconds: polling period in seconds to check
+    """
+
+    def __init__(
+        self,
+        bucket: str,
+        prefix: str,
+        polling_period_seconds: float,
+        google_cloud_conn_id: str,
+        hook_params: dict,
+    ):
+        super().__init__(
+            bucket=bucket,
+            object_name=prefix,
+            polling_period_seconds=polling_period_seconds,
+            google_cloud_conn_id=google_cloud_conn_id,
+            hook_params=hook_params,
+        )
+        self.prefix = prefix
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """
+        Serializes GCSBlobTrigger arguments and classpath.
```

Comment:
It should be `GCSPrefixBlobTrigger` instead of `GCSBlobTrigger`

Reply:
Done